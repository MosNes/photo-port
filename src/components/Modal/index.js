import React from 'react';

//onClose comes from props passed in from Photolist
function Modal({ currentPhoto, onClose }) {

    const { name, category, description, index } = currentPhoto;

    return (
        <div className='modalBackdrop'>
            <div className='modalContainer'>
                <h3 className='motalTitle'>{name}</h3>
                <img alt='current category' src={require(`../../assets/large/${category}/${index}.jpg`)} />
                <p>
                    {description}
                </p>
                <button type='button' onClick={onClose}>
                    Close this modal
                </button>
            </div>
        </div>
    );
}

export default Modal;