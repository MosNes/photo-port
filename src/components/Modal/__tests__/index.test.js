import React from "react";
import {render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();

const currentPhoto = {
    name: 'Park Bench',
    category: 'landscape',
    description: 'Lorem Ipsum dolor sit amet, consectetur adipiscing elit.',
    index: 1
}

afterEach(cleanup);

describe('Modal Component', () => {
    //render test
    it('renders', () => {
        render(<Modal 
            currentPhoto={currentPhoto}
            onClose={mockToggleModal}
            />);
    });

    //snapshot test
    it('matches snapshot', () => {
        const {asFragment} = render(
            <Modal 
            currentPhoto={currentPhoto}
            onClose={mockToggleModal}
            />
        )
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal 
            currentPhoto={currentPhoto}
            onClose={mockToggleModal}
            />);
            
        fireEvent.click(getByText('Close this modal'));

        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    })
})