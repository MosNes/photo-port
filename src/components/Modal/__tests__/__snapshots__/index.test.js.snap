// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="motalTitle"
      >
        Park Bench
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Lorem Ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
